test/extended/etcd/vertical_scaling.go

```diff
@@ -186,4 +197,95 @@ var _ = g.Describe("[sig-etcd][Feature:EtcdVerticalScaling][Suite:openshift/etcd
 		err = errors.Wrap(err, "scale-down: timed out waiting for only 3 Running master machines")
 		o.Expect(err).ToNot(o.HaveOccurred())
 	})
+
+	// The following test covers a basic vertical scaling scenario when CPMS is disabled
+	// and validates that the scale-down does not happen before the scale-up event.
+	// When CPMS is active, we can't ideally confirm that scale-down doesn't happen before scale-up because,
+	// the clustermemberremoval controller will race to remove the old member (from the machine pending deletion)
+	// as soon as the new machine's member is promoted to a voting member.
+	//
+	// 1) If the CPMS is active, first disable it by deleting the CPMS custom resource
+	// 2) Delete the machine
+	// 3) Create a new master machine that belongs to the same index as the deleted machine
+	// 4) Scale-up happens first before the scale-down
+	// 5) Validate scales-up happens first by verifying 4 voting members in etcd cluster
+	// 6) Then scale-down is validated by confirming the member removal and changes in the cluster membership
+	g.It("is able to vertically scale up and down when CPMS is disabled [apigroup:machine.openshift.io]", func(ctx context.Context) {
+		if cpmsActive {
+			// step 0: disable the CPMS
+			framework.Logf("Disable the CPMS")
+			err := scalingtestinglibrary.DisableCPMS(ctx, g.GinkgoT(), cpmsClient)
+			err = errors.Wrap(err, "pre-test: failed to disable the CPMS")
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			// re-enable CPMS after the test
+			defer func() {
+				framework.Logf("Re-enable the CPMS")
+				err := scalingtestinglibrary.EnableCPMS(ctx, g.GinkgoT(), cpmsClient)
+				err = errors.Wrap(err, "post-test: failed to re-enable the CPMS")
+				o.Expect(err).ToNot(o.HaveOccurred())
+			}()
+		}
+
+		framework.Logf("CPMS is disabled. The test will delete an existing machine and manually create a new machine to validate scale-down doesn't happen before scale-up event")
+
+		// step 1: delete a running machine
+		deletedMachineName, err := scalingtestinglibrary.DeleteSingleMachine(ctx, g.GinkgoT(), machineClient)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		framework.Logf("Deleted machine %q", deletedMachineName)
```

Comment:
```Go
// 2) Delete the machine
// 3) Create a new master machine that belongs to the same index as the deleted machine
// 4) Scale-up happens first before the scale-down
// 5) Validate scales-up happens first by verifying 4 voting members in etcd cluster
// 6) Then scale-down is validated by confirming the member removal and changes in the cluster membership
```

Conversely, I think we want to wait a bit and ensure that the scale-down has not already occurred by simply deleting a machine.
This is exactly what we regressed on in OCPBUGS-36462 when deleting a machine caused the scale-down controller to remove a member while we were still at 3 members.
https://issues.redhat.com/browse/OCPBUGS-36462?focusedId=25232151&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-25232151

I couldn't add that in the other test because CPMSO would just create the new machine. But now we can wait e.g 30 seconds, and then check that we're still at 3 members and no scale-down happening. Then we can move onto step 2 and add the new machine.